service/client/websocket.js

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+const MAX_WEBSOCKET_FAILS = 7;
+const MIN_WEBSOCKET_RETRY_TIME = 3000; // 3 sec
+const MAX_WEBSOCKET_RETRY_TIME = 300000; // 5 mins
+
+/* eslint-disable global-require, no-process-env */
+
+let Socket;
+if (process.env.NODE_ENV === 'test') {
+    Socket = require('ws');
+} else {
+    Socket = WebSocket;
+}
+
+/* eslint-enable global-require, no-process-env */
+
+class Websocket {
```

Comment:
Take a look at the bottom of the file, maybe you'll feel more confortable if we rename the file?